modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDeploymentResult.java

```diff
@@ -50,42 +50,40 @@ public void execute(DelegateExecution execution) {
 
         String processInstanceId = execution.getProcessInstanceId();
         Map<String, Object> variables = runtimeService.getVariables(processInstanceId);
-        UUID originalServiceId = (UUID) variables.get(RecreateConstants.ID);
-        log.info("Recreate workflow of Instance Id : {} check deploy service status with id:{}",
-                processInstanceId, originalServiceId.toString());
+        UUID serviceId = (UUID) variables.get(RecreateConstants.SERVICE_ID);
+        UUID recreateOrderId = (UUID) variables.get(RecreateConstants.RECREATE_ORDER_ID);
+        log.info("Recreate workflow of instance id : {} check deploy service status with id:{}",
+                processInstanceId, serviceId.toString());
 
-        ServiceRecreateEntity serviceRecreateEntity =
-                recreateService.getServiceRecreateEntityById(UUID.fromString(processInstanceId));
-        boolean isDeploySuccess =
-                deploymentEntityHandler.isServiceDeployedSuccess(originalServiceId);
+        boolean isDeploySuccess = deploymentEntityHandler.isServiceDeployedSuccess(serviceId);
         if (isDeploySuccess) {
-            recreateService.updateServiceRecreateStatus(serviceRecreateEntity,
-                    RecreateStatus.DEPLOY_COMPLETED, OffsetDateTime.now());
+            serviceOrderManager.completeOrderProgress(
+                    recreateOrderId, TaskStatus.SUCCESSFUL, null);
             runtimeService.setVariable(processInstanceId, RecreateConstants.IS_DEPLOY_SUCCESS,
                     true);
-            log.info("deployment step completed for recreate order {}", processInstanceId);
+            log.info("Deployment step completed for recreate order workflow with id {}.",
+                    processInstanceId);
         } else {
-            recreateService.updateServiceRecreateStatus(serviceRecreateEntity,
-                    RecreateStatus.DEPLOY_FAILED, OffsetDateTime.now());
             runtimeService.setVariable(processInstanceId, RecreateConstants.IS_DEPLOY_SUCCESS,
                     false);
-            int deployRetryNum = updateDeployRetryNum(processInstanceId, variables);
+            int deployRetryNum = getDeployRetryNum(variables);
+            runtimeService.setVariable(processInstanceId, RecreateConstants.DEPLOY_RETRY_NUM,
```

Comment:
Get the error message from the ServiceDeploymentEntity and set it as the value of  'RECREATE_ERROR_MESSAGE' into variables of the processIntanceId

Reply:
fixed
like we discussed, use deploymentEntityHandler.getServiceDeploymentEntity(serviceId) to query resultMessage.